### PR TITLE
Fixes #14831: Clean up rudder agent check

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/14831

### CODE REVIEW

**Comment by peckpeck** [comment]
Still some TODO

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

File: share/commands/agent-check [hunk 1]
```diff
@@ -260,63 +263,64 @@ check_varspace() {
...
-        stop_now=1
-      fi
+      check_space "${PG_TABLES_PATH}" 98
     fi
   fi
+
+  # If we are using TCP, we cannot stop postgresql otherwise all infra can go down - but we can stop rudder-jetty and disable rudder agent
+  use_tcp=0
+  NB_TCP=$(grep -r "TCP" /var/rudder/cfengine-community/inputs/promises.cf |  wc -l)
+  if [ 2 -eq "${NB_TCP}" ]; then
```
Comment (amousset) [inline share/commands/agent-check]: Isn't this present in rudder.json?

Comment (amousset) [reply share/commands/agent-check]: It is, could you use it instead? It will be much more robust.

File: share/commands/agent-check [hunk 2]
```diff
@@ -325,46 +329,22 @@ then
...
 fi
 
-# Launch each check with a certain order
+# Check for free space and stop if there is not enough
 check_varspace
+# Agent must have an UUID
 check_and_fix_rudder_uuid
 
-# Once for rudder's cfengine
-if [ -e "${RUDDER_CFE_DIR}/bin/cf-agent" ]
```
Comment (amousset) [inline share/commands/agent-check]: :open_mouth:

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**